### PR TITLE
Add remote linux glob behavior to expand paths from patterns

### PR DESCRIPTION
**Pull request by pevogam**
This was previously done via `ops.ls` but the `ls` would return a "serious error" exit code 2 when the expanded glob pattern results in zero paths. The current reimplementation makes use of `echo` to achieve the same with safe zero path outcome that won't hide more serious errors.

Signed-off-by: Plamen Dimitrov <plamen.dimitrov@intra2net.com>

### CODE REVIEW

**Comment by pevogam** [comment]
@ldoktor Do you think it is also possible to release 1.6.5 after this pull request is merged? This would greatly simplify deploying all the merged work from the past weeks / years on our side and would be really appreciated.

**Comment by pevogam** [comment]
@lmr Could you help review this and potentially release?

File: aexpect/ops_linux.py
Comment (ldoktor) [inline aexpect/ops_linux.py:236]: This will not work for paths containing spaces. (eg `touch "a b"`)

File: aexpect/ops_linux.py
Comment (ldoktor) [inline aexpect/ops_linux.py:237]: I liked this idea to ensure the path exists, but the cons is it's really slow. For 100 files it took 50s on my PC. Provided you fix the glob to really return the correct paths, this should not be needed, right?

File: aexpect/ops_linux.py
Comment (pevogam) [reply aexpect/ops_linux.py:236]: Hmmm this is indeed a limitation however, I would prefer to simply put a disclaimer in the docstring as there is no better solution I would prefer. As you can see in our original discussion the only other option is shell specific and I would prefer not to go in this direction. We already ruled out the `ls` hiding status and this is the only option that remains. Is this ok?

File: aexpect/ops_linux.py
Comment (pevogam) [reply aexpect/ops_linux.py:237]: We could then just fall back to the original `ls` solution hiding the status. Even though I had to investigate for solutions for not hiding the status and we end up back where we started which sounds like some wasted time and effort. Or do you have a better solution you could propose?

File: aexpect/ops_linux.py
Comment (ldoktor) [reply aexpect/ops_linux.py:236]: I'd probably use `MATCHES=({globpattern})\nfor MATCH in "${MATCHES[@]}"; do echo "$MATCH"; done` although I'm not 100% sure we won't end up with some weird chars issues as well.

Comment (pevogam) [reply aexpect/ops_linux.py:236]: This sounds pretty much like my comment https://github.com/avocado-framework/aexpect/pull/98#discussion_r988658483 which remains shell specific or am I missing something?

File: aexpect/ops_linux.py
Comment (ldoktor) [reply aexpect/ops_linux.py:237]: Looks like there is no perfect solution. There will always be some limitation. The `ls` solution should do in most cases s I think it makes sense to use that one and leave it up to people to use custom handling in case this fails.

Comment (pevogam) [reply aexpect/ops_linux.py]: Alright, pushed the changes just now, I think CI will pass.

File: aexpect/ops_linux.py
Comment (ldoktor) [inline aexpect/ops_linux.py:261]: In order to be able to split by lines it's necessary to enforce `-1` argument:

```
>>> aexpect.ops_linux.ls(a, "*", False, '-1UNq')
['a', 'a b']
>>> aexpect.ops_linux.ls(a, "*", False, '')
["a  'a b'"]
```

Also note that the `ls` approach can result in surprising output when in use with dirs:

```
>>> aexpect.ops_linux.ls(a, "/tmp/b", False, '-1')
['a', 'b']
>>> aexpect.ops_linux.ls(a, "/tmp/b/*", False, '-1')
['/tmp/b/a:', '', '/tmp/b/b:']
```

I don't want to draw the line here, it depends on which features you, as the primary user, need. I'm fine with any approach as long as it gets properly documented and is somehow systematic. So I'd suggest the use of glob and arrays, or ls with the usual arguments (perhaps just call the `ops_linux.ls` and catch for status 2?)

Comment (pevogam) [reply aexpect/ops_linux.py:261]: All good points and integration tests here are still running hence I didn't notice this yet. How about using something that will simplify all of these residues of user formatted output like `find`?
```
[pevogam@drogon androids]$ find c*
c
c b
[pevogam@drogon androids]$ ls c*
'c b'

c:
```
I guess it depends what the user would expect to get under glob but I would reckon receiving a guaranteed list of paths seems much more reasonable expectation than dir-ls-like behavior.

Comment (pevogam) [reply aexpect/ops_linux.py:261]: (the `-1` was indeed important but my testing didn't catch it because of some default `ls` settings on some linux distros)

Comment (ldoktor) [reply aexpect/ops_linux.py:261]: I though about find for a while but decided not to pursuit it further, but it can also work. Only I'd suggest using `-maxdepth 0`. With that it works well for all of the variants I tried.

Comment (ldoktor) [reply aexpect/ops_linux.py:261]: find /tmp/*/*a* -maxdepth 0

Comment (ldoktor) [reply aexpect/ops_linux.py:261]: find /tmp/b/* -maxdepth 0

Comment (pevogam) [reply aexpect/ops_linux.py:261]: Thanks and I apologize for the micromanagement here - pushed and CI passing, will need more time to confirm the integration tests we have but I think we are both feeling confident about the final solution now.

**Review by ldoktor — APPROVED** [review]
Thanks, seems to work well now

File: aexpect/ops_linux.py
Comment (ldoktor) [reply aexpect/ops_linux.py:261]: No problem, that is what reviews are for.

**Comment by pevogam** [comment]
> @ldoktor Do you think it is also possible to release 1.6.5 after this pull request is merged? This would greatly simplify deploying all the merged work from the past weeks / years on our side and would be really appreciated.

Thanks a lot @ldoktor! What are your plans regarding future release of aexpect? Do you think something like 1.6.5 will be possible anytime in the near future?

**Comment by ldoktor** [comment]
I don't have the rights to release aexpect, so the decision is on @lmr, anyway I'd appreciate to discuss the https://github.com/avocado-framework/aexpect/pull/114 first as this would be the first official release with the new libraries. So Lucas, would you please take a look a that?

**Pull request merged**
